schedulers/argo-workflow/addons.tf

```diff
@@ -40,30 +40,15 @@ module "eks_blueprints_kubernetes_addons" {
   # Apache YuniKorn Add-on
   #---------------------------------------------------------------
   enable_yunikorn = true
+  #---------------------------------------------------------------
+  # Argo Events Add-on
+  #---------------------------------------------------------------
+  enable_argo_workflows = true
 }
 
-module "helm_addon" {
-  source      = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons/helm-addon?ref=v4.12.0"
-  helm_config = local.default_helm_config
-  irsa_config = local.irsa_config
-
-  addon_context = {
-    aws_caller_identity_account_id = data.aws_caller_identity.current.account_id
-    aws_caller_identity_arn        = data.aws_caller_identity.current.arn
-    aws_eks_cluster_endpoint       = module.eks_blueprints.eks_cluster_endpoint
-    aws_partition_id               = data.aws_partition.current.partition
-    aws_region_name                = data.aws_region.current.name
-    eks_cluster_id                 = module.eks_blueprints.eks_cluster_id
-    eks_oidc_issuer_url            = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
-    eks_oidc_provider_arn          = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
-    tags                           = local.tags
-    irsa_iam_role_path             = "/"
-    irsa_iam_permissions_boundary  = ""
-  }
-}
 
 #---------------------------------------------------------------
-# Kubernetes Cluster role for argo workflows
+# Kubernetes Cluster role for argo workflows to run spark jobs
 #---------------------------------------------------------------
 resource "kubernetes_cluster_role" "spark-cluster" {
```

Comment:
Resource name should use underscore (`_`). e.g.,  `spark_cluster`. Same applies for all resource names and module names.

Reply:
May be change the resource name to `argo_workflows` and the metadata.name to `argo-workflows-role`

Reply:
changed to argo-workflows-spark-rolebinding
 and argo-workflows-admin-rolebinding